graphix/pattern.py

```diff
@@ -1653,7 +1680,11 @@ def measure_pauli(pattern, copy=False):
     vop_init = pattern.get_vops(conj=False)
     graph_state = GraphState(nodes=nodes, edges=edges, vops=vop_init)
     results = {}
-    to_measure, non_pauli_meas = pauli_nodes(pattern)
+    to_measure, non_pauli_meas = pauli_nodes(pattern, leave_inputs)
+    if leave_inputs:
+        new_inputs = pattern.input_nodes
+    else:
+        new_inputs = list(set(pattern.input_nodes) - set([i[0][1] for i in to_measure]))
```

Comment:
It might be clearer to set entire input_nodes to None, or replace the measured input index with None. This is because leaving only part of indices here makes it harder to have mapping between inputs and outputs. Also having non-equal number of inputs and outputs is potentially confusing.

I would suggest setting input_nodes to None if at least one of the input nodes are Pauli-measured. What's your thoughts @masa10-f

Reply:
That seems reasonable. I changed this part.